components/help/components/formatting.jsx

```diff
@@ -5,81 +5,851 @@ import React from 'react';
{', '}<code>{'cake'}</code>{', '}<code>{'cjsx'}</code>{', '}<code>{'cson'}</code>{', '}<code>{'iced'}</code>{', '}<code>{'cpp'}</code>{', '}<code>{'c'}</code>{', '}<code>{'cc'}</code>{', '}<code>{'h'}</code>{', '}<code>{'c++'}</code>{', '}<code>{'h++'}</code>{', '}<code>{'hpp'}</code>{', '}<code>{'cs'}</code>{', '}<code>{'csharp'}</code>{', '}<code>{'css'}</code>{', '}<code>{'d'}</code>{', '}<code>{'di'}</code>{', '}<code>{'dart'}</code>{', '}<code>{'delphi'}</code>{', '}<code>{'dpr'}</code>{', '}<code>{'dfm'}</code>{', '}<code>{'pas'}</code>{', '}<code>{'pascal'}</code>{', '}<code>{'freepascal'}</code>{', '}<code>{'lazarus'}</code>{', '}<code>{'lpr'}</code>{', '}<code>{'lfm'}</code>{', '}<code>{'diff'}</code>{', '}<code>{'django'}</code>{', '}<code>{'jinja'}</code>{', '}<code>{'dockerfile'}</code>{', '}<code>{'docker'}</code>{', '}<code>{'erl'}</code>{', '}<code>{'f90'}</code>{', '}<code>{'f95'}</code>{', '}<code>{'fsharp'}</code>{', '}<code>{'fs'}</code>{', '}<code>{'gcode'}</code>{', '}<code>{'nc'}</code>{', '}<code>{'go'}</code>{', '}<code>{'groovy'}</code>{', '}<code>{'handlebars'}</code>{', '}<code>{'hbs'}</code>{', '}<code>{'html.hbs'}</code>{', '}<code>{'html.handlebars'}</code>{', '}<code>{'hs'}</code>{', '}<code>{'hx'}</code>{', '}<code>{'java'}</code>{', '}<code>{'jsp'}</code>{', '}<code>{'js'}</code>{', '}<code>{'jsx'}</code>{', '}<code>{'json'}</code>{', '}<code>{'jl'}</code>{', '}<code>{'kt'}</code>{', '}<code>{'ktm'}</code>{', '}<code>{'kts'}</code>{', '}<code>{'less'}</code>{', '}<code>{'lisp'}</code>{', '}<code>{'lua'}</code>{', '}<code>{'mk'}</code>{', '}<code>{'mak'}</code>{', '}<code>{'md'}</code>{', '}<code>{'mkdown'}</code>{', '}<code>{'mkd'}</code>{', '}<code>{'matlab'}</code>{', '}<code>{'m'}</code>{', '}<code>{'mm'}</code>{', '}<code>{'objc'}</code>{', '}<code>{'obj-c'}</code>{', '}<code>{'ml'}</code>{', '}<code>{'perl'}</code>{', '}<code>{'pl'}</code>{', '}<code>{'php'}</code>{', '}<code>{'php3'}</code>{', '}<code>{'php4'}</code>{', '}<code>{'php5'}</code>{', '}<code>{'php6'}</code>{', '}<code>{'ps'}</code>{', '}<code>{'ps1'}</code>{', '}<code>{'pp'}</code>{', '}<code>{'py'}</code>{', '}<code>{'gyp'}</code>{', '}<code>{'r'}</code>{', '}<code>{'ruby'}</code>{', '}<code>{'rb'}</code>{', '}<code>{'gemspec'}</code>{', '}<code>{'podspec'}</code>{', '}<code>{'thor'}</code>{', '}<code>{'irb'}</code>{', '}<code>{'rs'}</code>{', '}<code>{'scala'}</code>{', '}<code>{'scm'}</code>{', '}<code>{'sld'}</code>{', '}<code>{'scss'}</code>{', '}<code>{'st'}</code>{', '}<code>{'styl'}</code>{', '}<code>{'sql'}</code>{', '}<code>{'swift'}</code>{', '}<code>{'tex'}</code>{', '}<code>{'vbnet'}</code>{', '}<code>{'vb'}</code>{', '}<code>{'bas'}</code>{', '}<code>{'vbs'}</code>{', '}<code>{'v'}</code>{', '}<code>{'veo'}</code>{', '}<code>{'xml'}</code>{', '}<code>{'html'}</code>{', '}<code>{'xhtml'}</code>{', '}<code>{'rss'}</code>{', '}<code>{'atom'}</code>{', '}<code>{'xsl'}</code>{', '}<code>{'plist'}</code>{', '}<code>{'yaml'}</code></span>
```

Comment:
This might actually be out of date since we periodically add languages. Instead of hardcoding all of them, maybe we could generate it from the list in `utils/constants.jsx` like
```jsx
import Constants from 'utils/constants.jsx';

const languages = [];
Objects.keys(Constants.HighlightedLanguages).forEach((language, i) => {
    if (i !== 0) {
        languages.push(<span key={language + '-divider'}>{, }</span>);
    }

    languages.push(<code key={language}>{language}</code>);
});
```